### PR TITLE
Update tweakreg documentation and minor API changes

### PR DESCRIPTION
**Pull request by mcara**
Update documentation for the ``tweakreg`` step.  This addresses PR https://github.com/STScI-JWST/jwst/issues/1523

### CODE REVIEW

File: docs/jwst/tweakreg/index.rst [hunk 1]
```diff
@@ -1,13 +1,25 @@
...
+**Also See:**
+
+.. toctree::
+   :maxdepth: 2
+
+   imalign
+   wcsimage
+   linearfit
+   matchutils
+   tweakreg_catalog
```
Comment (hbushouse) [inline docs/jwst/tweakreg/index.rst]: I don't think tweakreg_catalog and tweakreg_step exist anymore. They've been combined into the single tweakreg, which is what these docs are about. Or do they still exist as lower-level functions/modules?

File: jwst/tweakreg/__init__.py [hunk 2]
```diff
@@ -7,8 +7,8 @@
 __docformat__ = 'restructuredtext'
 
 __taskname__ = 'tweakreg'
-__version__ = '0.8.1'
-__vdate__ = '14-November-2017'
+__version__ = '0.8.2'
```
Comment (hbushouse) [inline jwst/tweakreg/__init__.py]: We should bump this up to 0.9.1, to keep in synch with the current development work that will be released as 0.9.x

File: jwst/tweakreg/matchutils.py [hunk 3]
```diff
@@ -1,3 +1,10 @@
+"""
+A module that provides algorithms for initial estimation of shifts
+based on 2D histograms.
+
+:License: `<http://www.stsci.edu/resources/software_hardware/pyraf/LICENSE>`_
```
Comment (hbushouse) [inline jwst/tweakreg/matchutils.py]: Is this the appropriate license reference to use? This isn't part of pyraf.

File: docs/jwst/tweakreg/index.rst (hunk 1 above)
Comment (mcara) [reply docs/jwst/tweakreg/index.rst]: @hbushouse I see both files in https://github.com/STScI-JWST/jwst/tree/master/jwst/tweakreg The new `tweakreg` step essentially combines the functionality of the two steps (old `tweakreg` and `tweakreg_catalog` steps). Probably `tweakreg_catalog.py` holds what was not moved to the common `tweakreg_step.py`. Maybe `tweakreg_catalog.py` should be renamed to simply `catalog.py`

File: jwst/tweakreg/__init__.py (hunk 2 above)
Comment (mcara) [reply jwst/tweakreg/__init__.py]: OK

File: jwst/tweakreg/matchutils.py (hunk 3 above)
Comment (mcara) [reply jwst/tweakreg/matchutils.py]: @hbushouse Oh, this is used almost everywhere in my contributed code - I carried it over from HST module "model". I will see how I can make this to point to the `LICENSE` file in the root directory.

**Comment by mcara** [comment]
@hbushouse Thanks!

**Pull request merged**